src/ProjectSchema/PropertyEditor.tsx

```diff
@@ -0,0 +1,173 @@
+import { FieldTitle } from "./FieldTitle";
+import { TypeIcon } from "../@shared";
+import * as css from "./PropertyEditor.css";
+
+import { SCHEMA_FIELD_TYPE, SchemaFieldType } from "../@types/schema";
+
+type PropertyOptions = {
+  min?: number;
+  max?: number;
+  required?: boolean;
+};
+
+type CurrentProperty = {
+  name: string;
+  type: SchemaFieldType;
+  options: PropertyOptions;
+};
+
+type Options = {
+  currentProperty: CurrentProperty;
+  setCurrentProperty: React.Dispatch<React.SetStateAction<CurrentProperty>>;
+};
+
+export function PropertyEditor({
+  currentProperty,
+  setCurrentProperty,
+}: Options) {
+  const handleEditProperty = ({
+    editType,
+    updateValue,
+  }: {
+    editType: string;
+    updateValue:
+      | React.ChangeEvent<HTMLInputElement>
+      | string
+      | boolean
+      | number;
+  }) => {
```

Comment:
```suggestion
  type EditableTypes =
    | { name: "name", editType: string }
    | { name: "type", editType: string }
    | { name: "required", editType: boolean }
    | { name: "min", editType: number }
    | { name: "max", editType: number };
   
  const handleEditProperty = <T extends EditableTypes["name"]>({
    editType,
    updateValue,
  }: {
    editType: T;
    updateValue: Extract<EditableTypes, { name: T }>["editType"]
  }) => {
```

간단하게 짜봤는데, 각 updateValue는 editType에 의해서 정해지니까 type assertion보다는 editType에 따라 지정되면 좋을 것 같아요~

Reply:
감사합니다 재혁님👏
정신없이 작업하다보니 놓친 부분이 많네요. dashboard 부분은 업데이트가 필요해서 contents, asset 작업하면서 함께 해보려 합니다. 삭제부분 보충하면서 추가 커밋 해놓을게요~

Reply:
넵넵 천천히 해주셔도 됩니다~~